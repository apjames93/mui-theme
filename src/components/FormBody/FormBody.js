import React from 'react';
import Grid from '@material-ui/core/Grid';
import PropTypes from 'prop-types';
import RenderForm from './components/RenderForm/RenderForm';
import Page from './components/Page/Page';
import styles from './FormBody.module.scss';

const FormBody = ({ form }) => (
  <Grid
    container
    direction="row"
    justify="flex-start"
    alignItems="flex-start"
    style={{ marginTop: '100px' }}
  >
    <Grid item xs={4} className={styles.container}>
      <RenderForm form={form} />
    </Grid>
    <Grid item xs={8} className={styles.container}>
      <Page form={form} />
    </Grid>
  </Grid>
);

export default FormBody;

FormBody.propTypes = {
  form: PropTypes.shape({
    getState: PropTypes.func,
  }),
};

FormBody.defaultProps = {
  form: {},
};
