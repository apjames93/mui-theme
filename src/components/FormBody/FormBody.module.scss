.container {
    overflow: scroll;
    max-height: 1000px;
}