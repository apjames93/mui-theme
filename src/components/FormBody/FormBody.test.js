import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import FormBody from './FormBody';

function setup() {
  const props = {
    form: {
      getState: jest.fn(() => ({
        values: {
          theme: {},
          storyBlokAccessToken: 'wasd',
        },
      })),
    },
  };
  const comp = mount(<FormBody {...props} />);
  return { comp, props };
}

describe('<FormBody />', () => {
  it('renders FormBody', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create((<FormBody {...props} />));
    expect(tree).toMatchSnapshot();
  });
});
