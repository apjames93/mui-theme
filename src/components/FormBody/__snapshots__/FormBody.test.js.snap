// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormBody /> snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start"
  style={
    Object {
      "marginTop": "100px",
    }
  }
>
  <div
    className="MuiGrid-root container MuiGrid-item MuiGrid-grid-xs-4"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-fullWidth"
      data-testid="Button"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        submit
      </span>
    </button>
  </div>
  <div
    className="MuiGrid-root container MuiGrid-item MuiGrid-grid-xs-8"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
      
    </div>
  </div>
</div>
`;
